src/mode.js

```diff
@@ -114,3 +130,37 @@ function parseQueryString_(queryString) {
   }
   return params;
 }
+
+/**
+ * Retrieve the `fullVersion` which will have a numeric prefix
+ * denoting canary/prod/experiment.
+ *
+ * @param {!Window} win
+ * @param {boolean} isLocalDev
+ * @return {string}
+ * @private
+ * @visibleForTesting
+ */
+export function getFullVersion_(win, isLocalDev) {
+  // If it's local dev then we won't actually have a full version so
+  // just use the version.
+  if (isLocalDev) {
+    return version;
+  }
+
+  // TODO(erwinm, #2917): Get full version from AMP_CONFIG if it exists.
+  if (win.AMP_CONFIG && win.AMP_CONFIG.fullVersion) {
+    return win.AMP_CONFIG.fullVersion;
+  }
+
+  const selector = 'script[src$="/v0.js"]';
```

Comment:
so right now we don't have `win.AMP_CONFIG.fullVersion` so this is to look for v0.js and read the version from the script tags src when the page is on the cache. I should implement #2917 by EOD. so you're right im not sure we need it. let me know what you think.